xwiki-platform-core/xwiki-platform-ratings/xwiki-platform-ratings-api/src/main/java/org/xwiki/ratings/events/UpdateAverageRatingFailedEvent.java

```diff
@@ -25,16 +25,28 @@
 import org.xwiki.stability.Unstable;
 
 /**
- * Event sent whenever an update of {@link AverageRating} failed.
- * The event is sent with the following informations:
- *   - source: the identifier of the {@link RatingsManager}
+ * Event sent whenever an update of {@link AverageRating} failed. The event is sent with the following information:
+ * <ul>
+ *   <li>source: the identifier of the {@link RatingsManager}</li>
+ *   <li>data: a {@link java.util.List} of updated {@link AverageRating}</li>
+ * </ul>
  *
  * @version $Id$
  * @since 12.9RC1
  */
 @Unstable
 public class UpdateAverageRatingFailedEvent extends AbstractAverageRatingEvent implements EndEvent
 {
+    /**
+     * Default constructor.
+     *
+     * @since 13.0RC1
```

Comment:
Don't we also want to fix XWIKI-17038 in 12.10.x ?

Reply:
And 12.6.x too I guess.

Reply:
> And 12.6.x too I guess.

The fix cannot be cherry-picked in 12.6.x since the Ratings module has been refactored in 12.9.

Reply:
Sadly yes. I'll need to keep the principle but adapt it for the older version of the code.

Reply:
So I think @tmortagne meant that you missed a since for 12.10.3.

Reply:
> So I think @tmortagne meant that you missed a since for 12.10.3.

Yes.

Reply:
Should I also set 12.6.x even though I will not be a straightforward cherry pick?

Reply:
> Should I also set 12.6.x even though I will not be a straightforward cherry pick?

Well  since it won't be the same code, no. Not even sure it worths it to perform the fix for 12.6.x TBH.

Reply:
> Should I also set 12.6.x even though I will not be a straightforward cherry pick?

The only criteria is: does this new API go 12.6.x or not.

Reply:
@manuelleduc sounds like the new since is still missing. And should be 13.1RC1 now.

Reply:
Updated to 13.1RC1 and 12.10.4